http/src/request/channel/reaction/create_reaction.rs

```diff
@@ -49,17 +49,53 @@ impl<'a> CreateReaction<'a> {
         }
     }
 
+    fn request(&self) -> Request {
+        Request::from(Route::CreateReaction {
+            channel_id: self.channel_id.0,
+            emoji: self.emoji.clone(),
+            message_id: self.message_id.0,
+        })
+    }
+
     fn start(&mut self) -> Result<()> {
-        self.fut.replace(Box::pin(self.http.verify(Request::from(
-            Route::CreateReaction {
-                channel_id: self.channel_id.0,
-                emoji: self.emoji.clone(),
-                message_id: self.message_id.0,
-            },
-        ))));
+        let request = self.request();
+
+        self.fut.replace(Box::pin(self.http.verify(request)));
 
         Ok(())
     }
 }
 
 poll_req!(CreateReaction<'_>, ());
+
+#[cfg(test)]
+mod tests {
+    use super::CreateReaction;
+    use crate::{
+        request::{channel::reaction::RequestReactionType, Request},
+        routing::Route,
+        Client,
+    };
+    use percent_encoding::{utf8_percent_encode, NON_ALPHANUMERIC};
+    use twilight_model::id::{ChannelId, MessageId};
+
+    #[test]
+    fn test_request() {
+        let client = Client::new("foo");
+
+        let emoji = RequestReactionType::Unicode {
+            name: String::from("\u{1f303}"),
+        };
+
+        let builder = CreateReaction::new(&client, ChannelId(123), MessageId(456), emoji);
+        let actual = builder.request();
+
+        let expected = Request::from(Route::CreateReaction {
+            channel_id: 123,
+            emoji: utf8_percent_encode("\u{1f303}", NON_ALPHANUMERIC).to_string(),
+            message_id: 456,
+        });
+
+        assert_eq!(actual.path_str, expected.path_str);
```

Comment:
we should be testing whether the format emoji function returns correct data here. please move the test to that file and assert equality between a correctly encoded emoji and the result of calling the function.

Reply:
I'm not sure how this would satisfy what was requested by @vivian - where would I get a Request::path_str? ref: https://github.com/twilight-rs/twilight/pull/685#issuecomment-759701651

Reply:
I'm sorry, in my head i assumed that we would be testing only the function. i was unaware this was a pattern.